cvat/apps/engine/media_extractors.py

```diff
@@ -429,74 +429,71 @@ def _has_frame(self, i):
 
         return False
 
-    def _decode(self, container):
-        frame_num = 0
-        for packet in container.demux():
-            if packet.stream.type == 'video':
+    def __iter__(self):
+        with self._get_av_container() as container:
```

Comment:
I should note that this doesn't completely fix the problem, because the container will only be closed if the generator reaches the end or if there's an exception raised within it. If the generator is simply abandoned by the calling code before it reaches the end, then it'll never gain control back, so it'll never close the container.

So a more thorough fix would be to track down all places where this `__iter__` is called and add calls to `generator.close`.

I'm not saying you should do it in _this_ PR (the problem has already existed before this PR), but it should be done at some point, otherwise there can still be subtle memory leaks.

Reply:
Ok, let's do it in another PR